server/jetstream_api.go

```diff
@@ -1168,22 +1172,75 @@ func (s *Server) jsStreamCreateRequest(sub *subscription, c *client, subject, re
 		// We do not require other stream to exist anymore, but if we can see it check payloads.
 		var exists bool
 		var maxMsgSize int32
-
 		if s.JetStreamIsClustered() {
 			js, _ := s.getJetStreamCluster()
 			if sa := js.streamAssignment(acc.Name, cfg.Mirror.Name); sa != nil {
 				maxMsgSize = sa.Config.MaxMsgSize
+				streamSubs = sa.Config.Subjects
 				exists = true
 			}
 		} else if mset, err := acc.lookupStream(cfg.Mirror.Name); err == nil {
 			maxMsgSize = mset.cfg.MaxMsgSize
+			streamSubs = mset.cfg.Subjects
 			exists = true
 		}
 		if exists && cfg.MaxMsgSize > 0 && maxMsgSize > 0 && cfg.MaxMsgSize < maxMsgSize {
 			resp.Error = &ApiError{Code: 400, Description: "stream mirror must have max message size >= source"}
 			s.sendAPIErrResponse(ci, acc, subject, reply, string(msg), s.jsonResponse(&resp))
 			return
 		}
+		if cfg.Mirror.External != nil {
+			if cfg.Mirror.External.DeliverPrefix != _EMPTY_ {
+				deliveryPrefixes = append(deliveryPrefixes, cfg.Mirror.External.DeliverPrefix)
+			}
+			if cfg.Mirror.External.ApiPrefix != _EMPTY_ {
+				apiPrefixes = append(apiPrefixes, cfg.Mirror.External.ApiPrefix)
+			}
+		}
+	}
+	if len(cfg.Sources) > 0 {
+		for _, src := range cfg.Sources {
+			if src.External == nil {
+				continue
+			}
+			if s.JetStreamIsClustered() {
+				js, _ := s.getJetStreamCluster()
+				if sa := js.streamAssignment(acc.Name, src.Name); sa != nil {
```

Comment:
I don't think you can call js.streamAssignment() like that (it needs locking)

Reply:
fixed